### PR TITLE
[Workspace] Refactor: workspace detail page header

### PR DESCRIPTION
**Pull request by yubonluo**
### Description

Workspace detail page header support description and button.

## Screenshot

image

## Changelog

 - refactor: [Workspace] Refactor: workspace detail page header

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/workspace/public/components/workspace_detail/workspace_detail.test.tsx [hunk 1]
```diff
@@ -198,24 +201,6 @@ describe('WorkspaceDetail', () => {
...
-    fireEvent.click(getByText('Cancel'));
-    expect(queryByText('Delete workspace')).toBeNull();
-    fireEvent.click(getByText('delete'));
-    const input = getByTestId('delete-workspace-modal-input');
-    fireEvent.change(input, {
-      target: { value: 'delete' },
-    });
-    const confirmButton = getByTestId('delete-workspace-modal-confirm');
-    fireEvent.click(confirmButton);
-  });
```
Comment (SuZhou-Joe) [inline src/plugins/workspace/public/components/workspace_detail/workspace_detail.test.tsx:217]: Is there any way that we can keep this unit test?

Comment (yubonluo) [reply src/plugins/workspace/public/components/workspace_detail/workspace_detail.test.tsx]: Due to the inability to render this delete button, we can only judge whether this test id exists on the page.

Comment (SuZhou-Joe) [reply src/plugins/workspace/public/components/workspace_detail/workspace_detail.test.tsx]: Actually the unit test is covering the delete flow for our features. I just feel a little bit uncomfortable that we delete it.

The worst case is that we can mock the HeaderControl component to execute the `run` function to popup the modal.

File: src/plugins/workspace/public/components/workspace_detail/workspace_detail.test.tsx [hunk 2]
```diff
@@ -119,6 +119,11 @@ const WorkspaceDetailPage = (props: any) => {
         },
       },
       dataSourceManagement,
+      navigationUI: {
+        HeaderControl: ({ controls }) => {
+          return controls?.[0].testId ?? null;
```
Comment (SuZhou-Joe) [inline src/plugins/workspace/public/components/workspace_detail/workspace_detail.test.tsx:124]:
```suggestion
          return controls.map(control => <button data-test-subj={control.testId} onClick={control.run}>{control.content}</button>)
```
The main point to fake this component is to ensure we have 100% test coverage on the delete modal.

**Review by SuZhou-Joe — APPROVED** [review]

File: src/plugins/workspace/public/components/workspace_detail/workspace_detail.test.tsx (hunk 1 above)
Comment (yubonluo) [reply src/plugins/workspace/public/components/workspace_detail/workspace_detail.test.tsx]: Sure, in order to cover the delete flow, I execute the `run` function to show the delete modal by `controls[0].run();`

File: src/plugins/workspace/public/components/workspace_detail/workspace_detail.test.tsx [hunk 3]
```diff
@@ -119,6 +119,14 @@ const WorkspaceDetailPage = (props: any) => {
         },
       },
       dataSourceManagement,
+      navigationUI: {
+        HeaderControl: ({ controls }) => {
+          if (props.showDeleteModal && controls && controls[0] && controls[0].run) {
+            controls[0].run();
+          }
```
Comment (AMoo-Miki) [inline src/plugins/workspace/public/components/workspace_detail/workspace_detail.test.tsx:126]:
```suggestion
          if (props.showDeleteModal) {
            controls?.[0]?.run?.();
          }
```

File: src/plugins/workspace/public/components/workspace_detail/workspace_detail.tsx [hunk 4]
```diff
@@ -146,25 +155,33 @@ export const WorkspaceDetail = (props: WorkspaceDetailProps) => {
...
+            setMountPoint={application.setAppDescriptionControls}
+          />
+        )}
+        <HeaderControl
+          controls={[
+            {
+              run: () => setDeletedWorkspace(currentWorkspace),
+              color: 'danger',
+              iconType: 'trash',
+              ariaLabel: 'Delete',
```
Comment (AMoo-Miki) [inline src/plugins/workspace/public/components/workspace_detail/workspace_detail.tsx:177]: Please use i18n.

File: src/plugins/workspace/public/components/workspace_form/workspace_detail_form.tsx [hunk 5]
```diff
@@ -34,7 +33,7 @@ interface FormGroupProps {
 const FormGroup = ({ title, children, describe }: FormGroupProps) => (
   <>
     <EuiFlexGroup gutterSize="xl">
-      <EuiFlexItem grow={false} className="workspace-detail-form-group">
+      <EuiFlexItem grow={false} style={{ width: '15%' }}>
```
Comment (AMoo-Miki) [inline src/plugins/workspace/public/components/workspace_form/workspace_detail_form.tsx:36]: Why do we need this?

File: src/plugins/workspace/public/components/workspace_initial/workspace_initial.tsx [hunk 6]
```diff
@@ -40,10 +40,14 @@ export const WorkspaceInitial = () => {
...
-      'Contact your administrator to create a workspace or to be added to an existing one.',
-  });
+  const noAdminToolTip = (
+    <>
+      {i18n.translate('workspace.initial.card.createWorkspace.toolTip', {
+        defaultMessage:
+          'Contact your administrator to create a workspace or to be added to an existing one.',
+      })}
+    </>
+  );
```
Comment (AMoo-Miki) [inline src/plugins/workspace/public/components/workspace_initial/workspace_initial.tsx:50]: OuiToolTip's `content` can take strings as part of RactNode.
```suggestion
  const noAdminToolTip = i18n.translate('workspace.initial.card.createWorkspace.toolTip', {
      defaultMessage:
        'Contact your administrator to create a workspace or to be added to an existing one.',
  });
```

File: src/plugins/workspace/public/components/workspace_initial/workspace_initial.tsx [hunk 7]
```diff
@@ -69,7 +73,7 @@ export const WorkspaceInitial = () => {
             defaultMessage: 'Create a workspace',
           })}
           description={
-            <EuiToolTip content={isDashboardAdmin || noAdminToolTip}>
+            <EuiToolTip content={isDashboardAdmin ? <></> : noAdminToolTip}>
```
Comment (AMoo-Miki) [inline src/plugins/workspace/public/components/workspace_initial/workspace_initial.tsx:76]: To disable the tooltip, you can set `null` or `undefined`
```suggestion
            <EuiToolTip content={isDashboardAdmin ? null : noAdminToolTip}>
```

File: src/plugins/workspace/public/components/workspace_form/workspace_detail_form.tsx (hunk 5 above)
Comment (yubonluo) [reply src/plugins/workspace/public/components/workspace_form/workspace_detail_form.tsx:36]: The width is to maintain the spacing between title and child ：
image
If there is no width：
image

File: src/plugins/workspace/public/components/workspace_initial/workspace_initial.tsx (hunk 7 above)
Comment (yubonluo) [reply src/plugins/workspace/public/components/workspace_initial/workspace_initial.tsx]: Thanks, have updated.

File: src/plugins/workspace/public/components/workspace_initial/workspace_initial.tsx (hunk 6 above)
Comment (yubonluo) [reply src/plugins/workspace/public/components/workspace_initial/workspace_initial.tsx]: Thanks, have updated.

File: src/plugins/workspace/public/components/workspace_detail/workspace_detail.tsx (hunk 4 above)
Comment (yubonluo) [reply src/plugins/workspace/public/components/workspace_detail/workspace_detail.tsx]: Thanks, have updated.

File: src/plugins/workspace/public/components/workspace_detail/workspace_detail.test.tsx (hunk 3 above)
Comment (yubonluo) [reply src/plugins/workspace/public/components/workspace_detail/workspace_detail.test.tsx]: Thanks, have updated.

**Review by SuZhou-Joe — APPROVED** [review]

**Review by ruanyl — APPROVED** [review]

**Pull request merged**